### PR TITLE
WIP: Pagination component base

### PR DESCRIPTION
**Pull request by francocorreasosa**
Had to create a PR to get a now.sh URL to share. Labeling as `don't merge` for the time being.

### Pagination
imagen

### Pagination Toolbar
image

### Pager
image

### CODE REVIEW

File: core/components/molecules/pagination/pagination-toolbar.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+```meta
```
Comment (siddharthkp) [inline core/components/molecules/pagination/pagination-toolbar.md]: I don't think we need to document `pagination-toolbar`, it's an internal component 

(p.s. it also breaks on the docs right now)

File: core/components/molecules/pagination/pagination.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+```meta
+  category: Layout
+  description: "Paginate resources with a handy pagination component"
+```
+
+```jsx
+<Pagination items={20372} perPage={10} page={3} />
```
Comment (siddharthkp) [inline core/components/molecules/pagination/pagination.md]: To make the prop table work, you need to pass `{props}` to the component and you can set the intended defaults with the `defaults` prop

Example:

```jsx
< Pagination {props} defaults={{items: 20372, perPage: 10, page: 3}} />
```


More here: https://auth0-cosmos-master.now.sh/docs/#/contribution-guide

File: core/components/molecules/pagination/utils.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+/**
+ * Calculates the page input width based on the characters
+ * it contains.
+ * @param {number} page
+ */
```
Comment (siddharthkp) [inline core/components/molecules/pagination/utils.js]: yay code comments! 💯

File: core/components/molecules/pagination/pagination-toolbar.md (hunk 1 above)
Comment (francocorreasosa) [reply core/components/molecules/pagination/pagination-toolbar.md]: Yep I know. Working on that, so should we completely remove that component from docs?

Comment (landitus) [reply core/components/molecules/pagination/pagination-toolbar.md]: > I don't think we need to document pagination-toolbar, it's an internal component

We need to show how to use this component somehow. The easiest way, for now, is to move it to our components list (not inside pagination, but as a standalone component).

Comment (siddharthkp) [reply core/components/molecules/pagination/pagination-toolbar.md]: Ah, you're right.

I assumed wrongly that `pagination-toolbar` is a child of `pagination`, it's a separate compound component

**Comment by francocorreasosa** [comment]
@siddharthkp @landitus Can you please take a look again to check if I'm missing something?

File: core/components/molecules/pagination-toolbar/pagination-toolbar.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    )
+  }
+}
+```
+
+### Example with totals
+
+You may want to show a little bit more detail of the content you are
+displaying, so we let you show the totals for the current page by passing the `showTotals`
+prop to the `<PaginationToolbar>` component.
```
Comment (siddharthkp) [inline core/components/molecules/pagination-toolbar/pagination-toolbar.md]: Not sure if that's the best name, but I can't think of a better one either 😅

`showIndex`,` showLegend`

@landitus how would you describe this component?

File: core/components/molecules/pager/pager.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      Older {'>'}
+    </Button>
+  </StyledPager>
+)
+
+Pager.propTypes = {
+  page: PropTypes.number.isRequired,
+  perPage: PropTypes.number.isRequired,
+  items: PropTypes.number.isRequired,
+  onPageChanged: PropTypes.func
```
Comment (siddharthkp) [inline core/components/molecules/pager/pager.js]: As an action, this should be `onPageChange` (drop the d at the end)

File: core/components/molecules/pager/pager.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+`
+
+const Pager = ({ onPageChanged, page, perPage, items }) => (
+  <StyledPager>
+    <Button
+      size="small"
+      appearance="secondary"
+      onClick={() => changePageIfAppropiate(page - 1, items, perPage, onPageChanged)}
+    >
+      {'<'} Newer
```
Comment (siddharthkp) [inline core/components/molecules/pager/pager.js]: We should replace the arrows with icons. You can achieve this with `chevron-down` + rotation if you feel like it

@landitus request for `chevron-left` + `chevron-right`

**Comment by siddharthkp** [comment]
The vertical alignment is slightly off

image

**Comment by siddharthkp** [comment]
This PR has a lot of good things, we should merge this and make the mentioned improvements in a different pull request

**Comment by siddharthkp** [comment]
Feature requests:

Can we merge the 3 styles of pagination introduced into this pull request as one component `[ Pager, Pagination Toolbar, Pagination ]`

For a person looking for a pagination component in cosmos, it would be confusing to pick the right one because they all have similar names (`Pager` and `Pagination` have the same description as well 😅)

Suggestion:

- In case of specialised components like `PaginationToolbar` which are built for tables, we should attach them to the table (`Table.Pagination`) instead of making them available as a new component.

- For components that are too similar, we should ask the question: Do we need both of them or can we pick one and call it the standard? @landitus need you here

**Pull request merged**